### DISCUSSION TITLE
Consequences of making a storage bucket public

### DISCUSSION THREAD

**Discussion by 0xda70d6** [supabase/supabase#3937] [Questions] [answered]
When creating a public bucket there's a warning saying:

"_Users can read objects in public buckets without any authorization. RLS policies are still required other operations such as object uploads and deletes._"

Does this mean that users can `select` from the bucket as well (and list items within the bucket), or just that items are accessible via the public URL?

My use case is that I want a bucket with files that are accessible through the public URL, but that uses (or anons) **cannot** do `listBuckets()` or `from.list()` to get a list of public buckets and/or files within the buckets.

While testing Supabase Storage it seems that the current behavior of public buckets matches my use case, but I just want to make sure, as it would  be very bad if anyone easily could get a list of all files within the public bucket.

**Comment by inian** [comment, marked as answer]
Only items are available via the public URL. All other api operations behave exactly like private buckets - so they need RLS policies to allow access to anon users

**Reply by 0xda70d6** [reply]
Perfect! Thanks for the quick reply!
